operator-common/src/main/java/io/strimzi/operator/common/metrics/MetricsHolder.java

```diff
@@ -5,37 +5,82 @@
  * Subclasses can add more specialized metrics.
  */
 public abstract class MetricsHolder {
-    protected static final String METRICS_PREFIX = "strimzi.";
+    /**
+     * Prefix used for metrics provided by Strimzi operators
+     */
+    public static final String METRICS_PREFIX = "strimzi.";
+    /**
+     * Metric name for number of reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS = METRICS_PREFIX + "reconciliations";
+    /**
+     * Metric name for number of periodic reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS_PERIODICAL = METRICS_RECONCILIATIONS + ".periodical";
+    /**
+     * Metric name for number of failed reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS_FAILED = METRICS_RECONCILIATIONS + ".failed";
+    /**
+     * Metric name for number of successful reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS_SUCCESSFUL = METRICS_RECONCILIATIONS + ".successful";
+    /**
+     * Metric name for duration of reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS_DURATION = METRICS_RECONCILIATIONS + ".duration";
+    /**
+     * Metric name for number of locked reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS_LOCKED = METRICS_RECONCILIATIONS + ".locked";
+    /**
+     * Metric name for number of resources managed by the operator.
+     */
+    public static final String METRICS_RESOURCES = METRICS_PREFIX + "resources";
+    /**
+     * Metric name for number of paused resources.
+     */
+    public static final String METRICS_RESOURCES_PAUSED = METRICS_RESOURCES + ".paused";
+    /**
+     * Metric name for certificate expiration timestamp in ms.
+     */
+    public static final String METRICS_CERTIFICATE_EXPIRATION_MS = METRICS_PREFIX + "certificate.expiration.ms";
 
     protected final String kind;
     protected final Labels selectorLabels;
     protected final MetricsProvider metricsProvider;
 
-    protected final Map<String, AtomicInteger> resourceCounterMap = new ConcurrentHashMap<>(1);
-    protected final Map<String, AtomicInteger> pausedResourceCounterMap = new ConcurrentHashMap<>(1);
-    private final Map<String, Counter> periodicReconciliationsCounterMap = new ConcurrentHashMap<>(1);
-    private final Map<String, Counter> reconciliationsCounterMap = new ConcurrentHashMap<>(1);
-    private final Map<String, Counter> failedReconciliationsCounterMap = new ConcurrentHashMap<>(1);
-    private final Map<String, Counter> successfulReconciliationsCounterMap = new ConcurrentHashMap<>(1);
-    private final Map<String, Counter> lockedReconciliationsCounterMap = new ConcurrentHashMap<>(1);
-    private final Map<String, Timer> reconciliationsTimerMap = new ConcurrentHashMap<>(1);
+    protected final Map<MetricKey, AtomicLong> certificateExpirationMap = new ConcurrentHashMap<>(1);
```

Comment:
Should this actually be here? Wouldn't some OperetorMetricsHolder be a better choice? (I'm not sure how easy or hard it would be to create some KafkaAssemblyOperatorMetricsHolder, that migth be too hard for this)

Reply:
If you like I'd give it a go, as I already thought of this - didn't do it as I thought I would be too big of a chance for the PR. But let me do a try 😄

Reply:
Pleas check my suggestion, think its alright @scholzj